packages/bathymetry/src/bathy.maker.ts

```diff
@@ -0,0 +1,218 @@
+        const hashPath = this.file.name(FileType.Hash);
+        if (fs.existsSync(hashPath)) return (await fs.promises.readFile(hashPath)).toString();
+        logger.info({ hashPath }, 'CreateHash');
+
+        const outputHash = await Hash.hash(this.filePath);
+        await fs.promises.writeFile(hashPath, outputHash);
+        return outputHash;
+    }
+
+    /** Render a specific tile */
+    async renderTile(tile: Tile, logger: LogType): Promise<void> {
+        try {
+            await this.createTile(tile, logger);
+            await this.createHillShadedTile(tile, logger);
+            await this.createCompositeTile(tile, logger);
+            await this.createTileMetadata(tile, logger);
+        } catch (err) {
+            logger.error({ err }, 'Failed');
+        }
+    }
+
+    /** convert a input file into one that can be processed by this script */
+    async createSourceGeoTiff(logger: LogType): Promise<void> {
+        if (fs.existsSync(this.tiffPath)) return;
+        logger.info({ path: this.tiffPath }, 'Converting to GeoTiff');
+        const gdal = Gdal.create();
+        await gdal.run(
+            'gdal_translate',
+            [
+                '-of',
+                'GTiff',
+                // Files need to be converted to Float32 to fix a weird outline bug with the resampling
+                // see https://github.com/linz/basemaps-team/issues/241
+                '-ot',
+                'Float32',
+                this.filePath,
+                this.tiffPath,
+            ],
+            logger,
+        );
+    }
+
+    /** Create a tile in the output TMS's CRS */
+    async createTile(tile: Tile, logger: LogType): Promise<void> {
+        const tileId = TileMatrixSet.tileToName(tile);
+        const warpedPath = this.file.name(FileType.Warped, tileId);
+        if (fs.existsSync(warpedPath)) return;
+
+        const bounds = this.tms.tileToSourceBounds(tile);
+        const warpCommand = [
+            '-of',
+            'GTIFF',
+            '-co',
+            'NUM_THREADS=ALL_CPUS',
+            '-s_srs',
+            Epsg.Wgs84.toEpsgString(),
+            '-t_srs',
+            this.tms.projection.toEpsgString(),
+            '-r',
+            'bilinear',
+            '-te',
+            ...bounds.toBbox(),
+            this.tiffPath,
+            warpedPath,
+        ].map(String);
+
+        const gdal = Gdal.create();
+        if (gdal.mount) gdal.mount(path.dirname(this.tiffPath));
+
+        logger.trace({ file: warpedPath }, 'Warping');
+        await gdal.run('gdalwarp', warpCommand, logger);
+    }
+
+    /** Create a hillshade for a tile */
+    async createHillShadedTile(tile: Tile, logger: LogType): Promise<void> {
+        const tileId = TileMatrixSet.tileToName(tile);
+        const warped = this.file.name(FileType.Warped, tileId);
+        const target = this.file.name(FileType.HillShade, tileId);
+
+        if (fs.existsSync(target)) return;
```

Comment:
Do we want to continue if path exists?

Reply:
this is a lazy processing thingy, if the file already exists skip the reprocessing of it, quite useful for testing it.